### ISSUE TITLE
Where filter not working.

### ISSUE THREAD

**Issue by KingsDevelopment** [strongloop/loopback-connector-redis#39]
My model:

    {
      "name": "Cart",
      "plural": "carts",
      "base": "PersistedModel",
      "idInjection": true,
      "options": {
        "validateUpsert": true
      },
      "mixins": {
        "Timestamp": true
      },
      "properties": {
        "customerId": {
          "type": "string",
          "required": true,
          "index": {
            "unique": true
          } 
        }
      },
      "validations": [],
      "relations": {
        "lines": {
          "type": "hasMany",
          "model": "CartLine",
          "foreignKey": "cartId"
        },
        "customer": {
          "type": "belongsTo",
          "model": "Customer",
          "foreignKey": "customerId"
        }
      },
      "acls": [],
      "methods": {}
    }

When doing:

     Cart
     .findOne({
          where: {
               customerId: "58147cab5185d50f002d5a70"
          }
     })

The result is `null`.

The record exists:
![records from explorer](https://puu.sh/s9ucV.png)

**Comment by KingsDevelopment** [comment]
Any idea? @superkhau @jannyHou

**Comment by superkhau** [comment]
Looks like a bug. Is the result null in your callback? 

```
Cart
 .findOne({
      where: {
           customerId: "58147cab5185d50f002d5a70"
      }
 }, function(err, carts) {
  console.log(carts);
});
```

**Comment by KingsDevelopment** [comment]
yap

**Comment by KingsDevelopment** [comment]
Any update?

**Comment by KingsDevelopment** [comment]
Clicked too fast..

**Comment by superkhau** [comment]
Can you provide a sample repo I can clone to reproduce on my end?

**Comment by KingsDevelopment** [comment]
sure, I will get to it. gimme a few seconds.

**Comment by KingsDevelopment** [comment]
So I had some kind of revelation, and not:

Steps:
- I created a new loopback project, installed redis & juggler connectos
- I copied my models from the current project. and removed acl's

Findings:
- It happened to be that the project I am working on running on an older version of juggler saved the id's as an escaped string (still does on timestamps), where as the where function would not work, because it was escaped.
- Creating a new entry in the new project with just the redis models, didn't escape the value, and then the where functionality did work on my project.

So afaik:
- Juggler update removed the escaping of the customerId (Did not check juggler project to confirm)
- It now works after updating... (we don't regulary update that fast, because we like to keep running)

So far, thanks for the help/dedication :+1:, But I guess it's fixed for now.
